editor/inserter/index.js

```diff
@@ -88,13 +94,18 @@ class Inserter extends wp.element.Component {
 }
 
 export default connect(
-	undefined,
+	( state ) => {
+		return {
+			selectedBlock: getSelectedBlock( state ),
```

Comment:
It seems to me that the inserter should have no knowledge of what block is selected. The inserter, in a way, only cares about inserting a block and dispatching the "show insertion point". Once we process the insert action the app would know the selected block internally and handle that aspect. What are your thoughts there?